wdl-engine/src/backend/local.rs

```diff
@@ -266,18 +280,35 @@ impl TaskExecution for LocalTaskExecution {
 /// Represents a task execution backend that locally executes tasks.
 ///
 /// This backend will directly spawn processes without using a container.
-#[derive(Debug, Default, Clone, Copy)]
-pub struct LocalTaskExecutionBackend;
+#[derive(Debug, Clone)]
+pub struct LocalTaskExecutionBackend {
+    /// The semaphore for handing out task execution permits.
+    lock: Arc<Semaphore>,
+    /// The maximum number of concurrent task executions.
+    max_concurrency: usize,
+}
 
 impl LocalTaskExecutionBackend {
     /// Constructs a new local task execution backend.
-    pub fn new() -> Self {
-        Self
+    ///
+    /// If `max_concurrency` is `None`, the default available parallelism for
+    /// the host will be used (typically the logical CPU count).
+    pub fn new(max_concurrency: Option<usize>) -> Self {
+        let max_concurrency =
+            max_concurrency.unwrap_or_else(|| available_parallelism().map(Into::into).unwrap_or(1));
```

Comment:
I'm wondering if this should warn if we have to hit the `unwrap_or(1)`.

Reply:
Certainly couldn't hurt; I'll add a warning.

Reply:
Yeah, my thought is just that, if this is happening, you probably want to be aware of it (though, I think it's very unlikely to happen on any platform we're targeting).